primitives/src/locktime/relative.rs

```diff
@@ -35,17 +36,35 @@ pub use units::locktime::relative::{Height, Time, TimeOverflowError};
 ///
 /// ```
 /// use bitcoin_primitives::relative;
+/// use bitcoin_primitives::BlockTime;
+/// use bitcoin_primitives::BlockHeight;
+/// use units::mtp_height::MtpAndHeight;
 /// let lock_by_height = relative::LockTime::from_height(144); // 144 blocks, approx 24h.
 /// assert!(lock_by_height.is_block_height());
 ///
 /// let lock_by_time = relative::LockTime::from_512_second_intervals(168); // 168 time intervals, approx 24h.
 /// assert!(lock_by_time.is_block_time());
 ///
-/// // Check if a lock time is satisfied by a given height or time.
-/// let height = relative::Height::from(150);
-/// let time = relative::Time::from_512_second_intervals(200);
-/// assert!(lock_by_height.is_satisfied_by(height, time));
-/// assert!(lock_by_time.is_satisfied_by(height, time));
+/// fn generate_timestamps(start: u32, step: u16) -> [BlockTime; 11] {
+///     let mut timestamps = [BlockTime::from_u32(0); 11];
+///     for (i, ts) in timestamps.iter_mut().enumerate() {
+///         *ts = BlockTime::from_u32(start.saturating_sub((step * i as u16).into()));
+///     }
+///     timestamps
+/// }
```

Comment:
I'd prefer the example to look more realistic. Nobody is going to generate the timestamps precedurally (but it's OK to have it as hidden function to replace the IO calls). So roughly:

```rust
let current_height = fetch_best_block_height();
let block_headers = core::array::try_from_fn(|i| fetch_block_header(current_height - 11 + i)?;
let block_times = block_headers.map(|header| header.block_time);
let current_mtp_and_height = MtpAndHeight::new(current_height, block_times);
let utxo_mined_at = load_utxo_from_db(utxo_out_point);
// perform the check here
```

Reply:
+1